### PR TITLE
Library cleanup & re-organization

### PR DESCRIPTION
**Pull request by drkameleon**
# Description

~~Mainly, the mess that has become the Converters module.~~ ⬅️ initial goal (lol)

- [x] Re-organize all library modules
   - [x] Make sure module-wide TODOs are at the top (and leave only function-specific ones interspersed)
   - [x] Separate functions, predicates (e.g. `integer?`) and constants (e.g. `true`) into different sections, pretty much like in our website (in many cases, having them all cramped up together makes it very difficult to spot what we are looking for; alphabetical sorting remains, only it will be per section, e.g.: all the functions together, sorted alphabetically, followed by the predicates, followed by the constants, and so on...)
   - [x] Clean up extra space between function definitions
- [x] Move misplaced functions to the appropriate module (not all, but at least the more obvious ones)
- [x] Create a new Types module (basically, the vast majority of functions defined in Reflection have to do with *types* and so do some from the Converters module, e.g. `to` - this one simply converts a given value from one type to another one)
- [x] Clean up the Converters module; the name is a complete misnomer: it contains functions that deal with types (and yes, *convert*ing things, e.g. `to` - but that doesn't justify the name at all), value generators/initializers (e.g. `array`, `dictionary`, `range`, `function`, etc) and some totally unrelated ones (e.g. `with` - this is neither a converter, nor a type-related function, it would normally belong to Core!)

So, that's a rough idea of what this PR is supposed to deal with.

Let's see if we can make it all a tiny bit cleaner and better-organized! 😉 🚀 

## Type of change

- [x] Code cleanup
- [x] Documentation (documentation-related additions)

### CODE REVIEW

File: src/vm/checks.nim [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            if unlikely(v.a.len < expected or v.a.len > maxExpected):
+                RuntimeError_IncompatibleBlockSize(currentBuiltinName, v.a.len, $(expected) & ".." & $(maxExpected))
+
+# TODO(VM/lib) verify implementation of `requireValue`
+#  particularly, the implementation of `pre` as a template and
+#  how it affects C-code generation
+#  labels: vm, enhancement, open discussion
+template requireValue*(v: Value, expected: set[ValueKind], position: int = 1, message: set[ValueKind] | string = {}) = 
+    when not defined(PORTABLE):
+        template pre(): untyped {.redefine.} =
```
[the remark this answers is not in the document]
Comment (drkameleon) [reply src/vm/checks.nim:112]: ???

[the remark this answers is not in the document]
Comment (drkameleon) [reply src/vm/checks.nim:112]: I still don't get this at all. The compilation works fine in every other workflow, except for Nimlint (or at least this particular run of Nimlint). Is this really an error?

**Comment by drkameleon** [comment]
I'm fairly satisfied with the work that has been done here, so... let's wrap this up and continue working on more specific issue by up-merging this PR.

So...

Ready to merge! 🚀

File: src/vm/checks.nim (hunk 1 above)
[the remark this answers is not in the document]
Comment (RickBarretto) [reply src/vm/checks.nim:112]: > Is this really an error?

Probably not. Nim LSP, Linter or suggest don't work well to me. A lot of false-positive hints and warnings...

This already happen to me before with the CI. What I can say is: trust the compiler instead.

**Pull request merged**